src/poh_recorder.rs

```diff
@@ -38,15 +46,49 @@ impl PohRecorder {
         max_tick_height: Option<u64>,
     ) -> Self {
         let poh = Arc::new(Mutex::new(Poh::new(last_entry_id, tick_height)));
-        PohRecorder {
+        PohRecorderLeader {
             poh,
             bank,
             sender,
             max_tick_height,
         }
     }
 
-    pub fn hash(&self) -> Result<()> {
+    fn check_max_tick_height_reached(&self, poh: &Poh) -> bool {
```

Comment:
Can you replace `check` with `is`? An `is` prefix has been our convention for functions that return a boolean. `has` is also acceptable whenever it is grammatically incorrect to use `is`.